src/ksc/Ksc/SUF/AD.hs

```diff
@@ -230,6 +235,13 @@ sufFwdRevPass gst subst = \case
   -- (vs is the list of variables that occur free in each branch of
   -- the if.  N.B. the Single Use Property implies that the free
   -- variable set is the same for each branch.)
+  --
+  -- N.B. Until we have proper sum types we use products with a dummy
```

Comment:
Maybe a link to an issue describing sum types

Reply:
Good idea. Do we have one already? I couldn't find one in search (below), which surprised me.

https://github.com/microsoft/knossos-ksc/search?p=1&q=sum&type=issues